File: speed.py
import math
def reverse(seq): 
    count = 0
    neg_count = -1
    length = len(seq) / 2
    if len(seq) % 2 == 0:
        while count != length:
            seq[count], seq[neg_count] = seq[neg_count],seq[count]
            count += 1
            neg_count -= 1
        return seq
    if len(seq) % 2 != 0:
        length = math.ceil(length)
        while count != length:
            seq[count], seq[neg_count] = seq[neg_count],seq[count]
            count += 1
            neg_count -= 1
        return seq

File: speed.js
function reverse(arr) {
    let count = 0;
    let negCount = 1;
    let len = arr.length / 2;
    if (arr.length % 2 == 0){
      while (count != len){
        let first = arr[count];
        arr[count] = arr[arr.length - negCount];
        arr[arr.length - negCount] = first;
        count++;
        negCount++;
      }
      return arr;
    }else if (arr.length % 2 != 0){
      let newLen = Math.ceil(len);
      while (count != newLen){
        let first = arr[count];
        arr[count] = arr[arr.length - negCount];
        arr[arr.length - negCount] = first;
        count++;
        negCount++;
      }
      return arr;
    }
 }